src/constraints/util/operators.ts

```diff
@@ -38,3 +38,8 @@ export interface Comparator {
 	(a: number, b: number): boolean;
 	(a: bigint, b: bigint): boolean;
 }
+
+export interface UrlOptions {
+	allowedProtocols?: string[];
+	allowedDomains?: string[];
```

Comment:
Should require `.` to be contained so invalid domains can't be passed, make this a `StringDomain` type and export it:
```suggestion
	allowedDomains?: `${string}.${string}`[];
```